core/vtk/ttkProjectionFromField/ttkProjectionFromField.h

```diff
@@ -58,20 +57,40 @@ class TTKPROJECTIONFROMFIELD_EXPORT ttkProjectionFromField
   vtkSetMacro(UseTextureCoordinates, bool);
   vtkGetMacro(UseTextureCoordinates, bool);
 
+  vtkSetMacro(ProjectPersistenceDiagram, bool);
+  vtkGetMacro(ProjectPersistenceDiagram, bool);
+
 protected:
   ttkProjectionFromField();
 
-  ~ttkProjectionFromField() override;
-
   int FillInputPortInformation(int port, vtkInformation *info) override;
 
   int FillOutputPortInformation(int port, vtkInformation *info) override;
 
+  /**
+   * @brief Switch a given Persistence Diagram representation
+   */
+  int projectPersistenceDiagram(vtkUnstructuredGrid *const inputDiagram,
```

Comment:
I believe the ProjectionFromField filter should be completely agnostic to what it's projecting.
it should have no explicit relation to persistence diagrams at all.

if the users want to project a persistence diagram, I believe the right use case should be:
- the users check a box in ProjectionFromField for using a pre-defined coordinate array ("Use a pre-defined coordinate array")
- a drop-down menu appears, listing all PointData arrays with 3 components
- the users select the array they want (for persistence diagrams, the one named "Coordinates").

then the rest of the code should just modify the coordinates of the point based on that array and that is it.
what do you think?

Reply:
In this case, the diagram diagonal would still be present in the spatial embedding and there would not be any `Birth` and `Death` array, confusing the filters that consume Persistence Diagrams (BottleneckDistance, PersistenceDiagramClustering, …)